docs/modules/concepts/pages/configuration.adoc

```diff
@@ -0,0 +1,34 @@
+= Configuration
+
+== Overview
+
+An important aspect to understand about James is the idea of "configuration".
+Sometimes "configuration" means different things to different people, so we
+wanted to ensure that what we mean by the concept is clear and precise, 
+because in order operate a James server, you will necessarily need to 
+configure it.
+
+A "*Configuration*" is a https://martinfowler.com/bliki/ValueObject.html[value object]
+that communicates to a James server how the xref:user/index.adoc#_operator[Operator] wishes the
+server to behave. It is very similar to an interface, but yet it is not quite an interface
+(which is why we call it something different).
+
+An "interface" provides a means for an Operator to interact with a server as part of the
+normal course of ongoing operations. Interfacing with the system essentially means either
+reading the server's current state, or updating the server's current state. A typical example
+would be adding a new mailbox via the xref:concepts:???.adoc[Useradmin API]. Each time the Operator
+interacts with the system via an interface, although the *state* of the system may change, 
+the *behavior* of the system is not expected to change.
+
+By contrast, "configuration" of a system (also a noun, but yet not the same as the
```

Comment:
sometimes the difference is artificial: for example you can, with some implementation, define what are your domains in the configuration. For others, it's part of the server state.

Reply:
Perhaps, but allow me to dig in to your example.

I would argue that the state of the domains (i.e. the list of domains that are in the system) can be seeded by an "initial list". This initialDomainList is indeed a configuration, as you mention, but it does not represent the state. Rather, the state gets initialized with the initialDomainList.

So although they are related, in my mind the different roles are quite clear. The domainList itself is the state, while the initialDomainList is just a configuration that seeds the state.

But I think it's good to test this concept further.

Perhaps you have other examples that may invalidate or challenge this concept?

Reply:
I have a good one that breaks the initial vs living state view: Look at the wonderful configuration entry :
```
# Use to distribute the emails of a given slice within your cassandra cluster
# A good value is 2*cassandraNodeCount
# This parameter can only be increased.
# Optional, default 1
mailqueue.view.bucketCount=1
```
This configuration will change the state of the server by adding buckets. You can increase but never decrease this count.

To implement that we have to keep track of the previous values and check if an invariant is broken when this configuration key change.

We could very well have implemented that we a default value / REST API couple and it would still be configuration.

Reply:
Yeah, that's a good one. So does this invalidate the concept? Or should we consider this as an edge case and live with it?